drivers/sensor/st/lis2de12/lis2de12.c

```diff
@@ -25,6 +25,483 @@ LOG_MODULE_REGISTER(LIS2DE12, CONFIG_SENSOR_LOG_LEVEL);
 
 static const uint16_t lis2de12_odr_map[10] = { 0, 1, 10, 25, 50, 100, 200, 400, 1620, 5376};
 
+#ifdef CONFIG_SENSOR_LOG_LEVEL_DBG
+static uint8_t reg_buffer[200];
+
+int lis2de12_print_registers(const struct device *dev)
+{
+
+	const struct lis2de12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	LOG_DBG("LIS2DE12 Register Dump:");
+
+	uint8_t reg = 0;
+	uint8_t reg_value = 0;
+
+	reg = LIS2DE12_STATUS_REG_AUX;
+	if (lis2de12_read_reg(ctx, reg, &reg_value, 1) < 0) {
+		LOG_ERR("Failed to Register 0x%02X", reg);
+		return -EIO;
+	}
+	LOG_DBG("REG:0x%02X, %s : 0x%02X :\n\tTOR=%s\n\tTDA=%s", LIS2DE12_STATUS_REG_AUX,
+		"LIS2DE12_STATUS_REG_AUX", reg_value,
+		((reg_value) & (1 << (7 - 2)) ? true : false)
+			? "new temperature has overwritten the previous data"
+			: "no overrun has occurred",
+		((reg_value) & (1 << (7 - 5)) ? true : false)
+			? "new temperature is available"
+			: "new temperature data is not yet available");
```

Comment:
```suggestion
	LOG_DBG("REG:0x%02X, %s : 0x%02X :", LIS2DE12_STATUS_REG_AUX, "LIS2DE12_STATUS_REG_AUX", reg_value);
	LOG_DBG("\tTOR=%s",
		((reg_value) & (1 << (7 - 2)) ? true : false)
			? "new temperature has overwritten the previous data"
			: "no overrun has occurred");
	LOG_DBG("\tTDA=%s",
		((reg_value) & (1 << (7 - 5)) ? true : false)
			? "new temperature is available"
			: "new temperature data is not yet available");
```

Der output sieht komisch aus:

![image](https://github.com/user-attachments/assets/6e6a25d1-f334-4dc1-9535-82f0197be4fe)

Das liegt am fehlenden `\r`. Aber am besten einfach niemals eine `\n` in log messages verwenden.


Ich würde aber auch \r\n statt \n akzeptieren.